bcda/api/alr.go

```diff
@@ -0,0 +1,132 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/CMSgov/bcda-app/bcda/models"
+	"github.com/CMSgov/bcda-app/bcda/models/postgres"
+	"github.com/CMSgov/bcda-app/bcda/responseutils"
+	"github.com/CMSgov/bcda-app/bcda/service"
+	"github.com/CMSgov/bcda-app/bcda/servicemux"
+	"github.com/CMSgov/bcda-app/bcda/web/middleware"
+	fhircodes "github.com/google/fhir/go/proto/google/fhir/proto/stu3/codes_go_proto"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func (h *Handler) alrRequest(w http.ResponseWriter, r *http.Request, reqType service.RequestType) {
+	ctx := r.Context()
+
+	ad, err := readAuthData(r)
+	if err != nil {
+		panic("Auth data must be set prior to calling this handler.")
+	}
+
+	rp, ok := middleware.RequestParametersFromContext(ctx)
+	if !ok {
+		panic("Request parameters must be set prior to calling this handler.")
+	}
+
+	if err := h.validateRequest(rp.ResourceTypes); err != nil {
+		oo := responseutils.CreateOpOutcome(fhircodes.IssueSeverityCode_ERROR, fhircodes.IssueTypeCode_EXCEPTION, responseutils.RequestErr,
+			err.Error())
+		responseutils.WriteError(oo, w, http.StatusBadRequest)
+		return
+	}
+
+	req := service.DefaultAlrRequest
+	if reqType == service.Runout {
+		req = service.RunoutAlrRequest
```

Comment:
Would something break if we matched the Runout # for ALR and BFD?

Reply:
Good call!